operator/cmd/mesh/profile-diff.go

```diff
@@ -75,7 +79,10 @@ func profileDiff(cmd *cobra.Command, rootArgs *rootArgs, pfArgs *profileDiffArgs
 		cmd.Println("Profiles are identical")
 	} else {
 		cmd.Printf("The difference between profiles:\n%s", diff)
-		os.Exit(1)
+		// The command was a success, but we return an error so that the process will
+		// exit with 1.  We don't want to tell the user there was an error because there wasn't.
+		cmd.SilenceErrors = true
+		return fmt.Errorf("dummy")
```

Comment:
what i had in mind more to make this more testable is to create a wrapper function with the existing signature and an internal function which could take an extra parameter (bool for exit?). then you could call it like this:
<inside profileDiff>
 err, doExit := doProfileDiff()..
 if doExit { os.Exit(1) }
Hope that makes sense - we can chat directly if it doesn't.
I think that would be a cleaner solution than returning dummy errors.

Reply:
OK, I have refactored so that it works as suggested.

The changes I made somehow seem to have also fixed https://github.com/istio/istio/issues/30375 .  That issue also has a PR, and the two PRs may conflict.